File: javascript/problem022.js
var fs = require('fs');

var data = fs.readFileSync('../textfiles/problem022.txt', 'utf8');
var nameArray = data.split(',');
var array = nameArray.sort();

function getScore(name, order){
  var score = 0;
  for(a=1; a<name.length-1; a++){
    score = score + name.charCodeAt(a) - 64;
  }
  return score * order;
}

var result = 0;
var index = 0;

for(i=0; i<array.length; i++){
  index = index + 1;
  result = result + getScore(array[i], index);
}

console.log(result);

File: python/problem022.py
f = open("../textfiles/problem022.txt", 'r')
data = f.read()
f.close()
data = data.replace("\"", "")
name_array = data.split(',')
name_array.sort()


def get_score(name, order):
    score = 0
    for a in name:
        score += ord(a) - 64
    return score * order


result = 0
index = 0
for i in name_array:
    index += 1
    result += get_score(i, index)
print(result)
